### DISCUSSION TITLE
Most efficient way to get sub-mesh from mesh using subset of vertices

### DISCUSSION THREAD

**Discussion by alpha027** [pyvista/pyvista#2668] [Q&A] [answered]
Hi,

The problem I have is the following: 
I have a big mesh object (around ~10 M vertices) and I want to extract small sub-meshes from it, the sub-meshes are not necessarily disjoint, but they are small in size (5% of the original mesh size). I use the method `remove_points(indices_to_remove)` for that. The thing is that for each sub-mesh I have a list of indices to remove, since the sub-meshes are small, the indices to remove list size gets very high and therefore leads to memory issues. Also when I run the code below, it takes around 1 second to get each sub-mesh.
I wonder if there is another more efficient way to extract sub-meshes from a mesh by selecting only the vertex indices to keep.

The code I have is something like this:
```python
submeshes = {}
for k, v in to_remove_indices.items():
        to_remove_points = v
        sub_mesh, ridx = mesh.remove_points(to_remove_points)
       # current submesh related to segment k
        submeshes[k] = sub_mesh
```

**Comment by whophil** [comment, marked as answer]
Have you tried [extract_points()](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.extract_points.html)?
